packages/syft/src/syft/service/worker/worker_pool.py

```diff
@@ -78,12 +117,85 @@ class WorkerPool(SyftObject):
     __version__ = SYFT_OBJECT_VERSION_1
 
     __attr_unique__ = ["name"]
-    __attr_searchable__ = ["name", "syft_worker_image_id"]
+    __attr_searchable__ = ["name"]
+    __repr_attrs__ = [
+        "name",
+        "image",
+        "max_count",
+        "workers",
+        "created_at",
+    ]
 
     name: str
-    syft_worker_image_id: UID
+    image: Optional[SyftWorkerImage]
     max_count: int
     workers: List[SyftWorker]
+    created_at: DateTime = DateTime.now()
+
+    @property
+    def running_workers(self) -> List[SyftWorker]:
+        with contextlib.closing(docker.from_env()) as client:
+            return [
+                worker
+                for worker in self.workers
+                if _get_worker_container_status(client, worker) == WorkerStatus.RUNNING
+            ]
+
+    @property
+    def healthy_workers(self) -> List[SyftWorker]:
+        with contextlib.closing(docker.from_env()) as client:
```

Comment:
Same here, this should be triggered over an API call instead.

Reply:
Thanks @shubham3121, done!